src/v/pandaproxy/server.cc

```diff
@@ -85,6 +116,12 @@ struct handler_adaptor : ss::httpd::handler_base {
         auto measure = _probe.auto_measure();
         auto guard = gate_guard(_pending_requests);
         server::request_t rq{std::move(req), this->_ctx};
+        if (_require_auth) {
+            const auto& cred_store = _controller->get_credential_store().local();
+            authenticate(*rq.req, cred_store);
+        }
+
+        vlog(plog.info, "PANDA authn success.");
```

Comment:
So the authn is a success but I'm getting failures when control gets to the kafka subsystem likely because I need to pass the SASL creds into kafka. I just don't know the full path for how to do that.

There errors I'm seeing are:
```
INFO  2022-08-18 17:12:23,773 [shard 0] kafka/client - client.h:98 - Nyalia in client dispatch # <-- I added this log
WARN  2022-08-18 17:12:23,973 [shard 0] kafka/client - client.cc:157 - broker_error: kafka::client::broker_error ({ node: -1 }, { error_code: broker_not_available [8] })
INFO  2022-08-18 17:12:23,974 [shard 0] kafka/client - broker.cc:41 - connected to broker:-1 - localhost:9092
INFO  2022-08-18 17:12:23,974 [shard 0] kafka/client - broker.h:63 - Nyalia in broker dispatch # <-- I added this log
WARN  2022-08-18 17:12:23,974 [shard 0] kafka - connection_context.cc:376 - Error processing request: kafka::kafka_authentication_exception (requests.cc:168 - Unexpected request during authentication: 3)
INFO  2022-08-18 17:12:24,175 [shard 0] kafka/client - broker.cc:41 - connected to broker:-1 - localhost:9092
INFO  2022-08-18 17:12:24,175 [shard 0] kafka/client - broker.h:63 - Nyalia in broker dispatch # <-- I added this log
WARN  2022-08-18 17:12:24,175 [shard 0] kafka - connection_context.cc:376 - Error processing request: kafka::kafka_authentication_exception (requests.cc:168 - Unexpected request during authentication: 3)
```
which are coming from requests.cc::handle_auth (via process_request) but I don't see the connection between PP and handle_auth.
I traced the callstack into src/v/kafka/client/broker.h::dispatch but things fall off from there.

Reply:
Pandaproxy currently shares the broker set between all incoming connections. In order to propagate the principal, you'll need a new broker connection to Redpanda. And that broker connection will need the incoming auth credentials mapped as sasl credentials.

Reply:
It's probably worth spinning up a new kafka/client for each incoming request, keyed by the user (but with some caching and timeouts to cleanup)